### PR TITLE
Add shutdown endpoint [ECR-692]

### PR DESCRIPTION
**Pull request by DarkEld3r**

### CODE REVIEW

**Comment by DarkEld3r** [comment]
It "works" because the node shuts down, but I get the following error:
> thread 'main' panicked at 'Node return error: Error { repr: Custom(Custom { kind: Other, error: StringError("An error in the `Network` thread occurred: can\'t cancel routine") }) }'

Additionally, I'm not happy with events processing.

File: exonum/src/api/private/system.rs [hunk 1]
```diff
@@ -12,12 +12,13 @@
...
 
-use std::net::SocketAddr;
-use std::collections::HashMap;
-
+use serde_json;
 use router::Router;
 use iron::prelude::*;
 
+use std::net::SocketAddr;
+use std::collections::HashMap;
```
Comment (matklad) [inline exonum/src/api/private/system.rs]: Not sure there's a real rule for  it, but it seems that std imports are usually first?

File: exonum/src/events/internal.rs [hunk 2]
```diff
@@ -68,6 +65,19 @@ impl InternalPart {
                             .map_err(|_| panic!("Can't execute jump to round"));
                         to_box(fut)
                     }
+                    InternalRequest::Shutdown => {
+                        let internal_tx = internal_tx.clone();
+                        let fut = Ok(())
+                            .into_future()
+                            .and_then(move |_| {
```
Comment (matklad) [inline exonum/src/events/internal.rs]: Why do we need `Ok(()).and_then`? This should be equivalent to `futures::lazy`?

File: exonum/src/api/private/system.rs (hunk 1 above)
Comment (DarkEld3r) [reply exonum/src/api/private/system.rs]: Nope: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions
:smile:

**Comment by DarkEld3r** [comment]
@alekseysidorov Nice idea! I will update tests as a separate pull request: https://jira.bf.local/browse/ECR-847 :wink:

**Pull request merged**